pkg/vm/emit/emit.go

```diff
@@ -32,20 +33,35 @@ func Bool(w *io.BinWriter, ok bool) {
 	Opcode(w, opcode.PUSHF)
 }
 
+func padRight(s int, buf []byte) []byte {
+	if len(buf) == s {
+		return buf
+	}
+	newBuf := make([]byte, s)
+	copy(newBuf, buf)
+	return newBuf
+}
+
 // Int emits a int type to the given buffer.
 func Int(w *io.BinWriter, i int64) {
 	switch {
 	case i == -1:
 		Opcode(w, opcode.PUSHM1)
-	case i == 0:
-		Opcode(w, opcode.PUSHF)
-	case i > 0 && i < 16:
+	case i >= 0 && i < 16:
 		val := opcode.Opcode(int(opcode.PUSH1) - 1 + int(i))
 		Opcode(w, val)
 	default:
-		bInt := big.NewInt(i)
-		val := IntToBytes(bInt)
-		Bytes(w, val)
+		buf := IntToBytes(big.NewInt(i))
+		l := len(buf)
+		if l > 32 {
+			w.Err = errors.New("too big integer")
+			return
+		}
+
+		// l != 0 becase of switch
+		padSize := byte(8 - bits.LeadingZeros8(byte(l-1)))
+		Opcode(w, opcode.Opcode(padSize))
+		w.WriteBytes(padRight(1<<padSize, buf))
```

Comment:
Creating yet another buffer in `padRight` doesn't look right. We can either use some `[32]byte` here (as we know that it won't exceed this length) and copy to/slice from it or just enlarge the `buf`.

Reply:
Fixed.